### PR TITLE
Feat: add announcement active fields

### PR DESCRIPTION
**Pull request by alexVasylenko**

### CODE REVIEW

**Comment by czgu** [comment]
Can you put some screenshots and add some description for this feature?

**Comment by alexVasylenko** [comment]
> Can you put some screenshots and add some description for this feature?


So, the main idea for these changes is to add a possibility to schedule announcements. 

A user can set up the schedule and in that case, all announcements will be filtered by date, but without these parameters, it should work like before.

Also, I added the select for envs. My solution is based on reg_ex functionality The user can select env and improve this reg_ex because I don't want to break the previous functionality.

This is the video of how it works now.


https://user-images.githubusercontent.com/59963571/140512886-b968713f-84d7-4730-9055-1e205ac9e961.mov


https://user-images.githubusercontent.com/59963571/140512856-f0268862-ee08-40cb-98c4-518c880ea4ff.mov

https://user-images.githubusercontent.com/59963571/140512805-a40437ac-7f73-4b79-819a-6487e48e3c90.mov


https://user-images.githubusercontent.com/59963571/140513053-76b84d86-1808-411f-8b09-166efe0bbab3.mov

File: querybook/webapp/components/AppAdmin/components/DatePickerField/DatePickerField.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+const StyledDatePicker = styled(DatePicker)`
+    max-width: 100px;
+`;
+
+const StyledLabel = styled.label`
+    white-space: nowrap;
+    padding-right: 12px;
+`;
+
+function formatDate(dateInSeconds?: number): Date {
```
Comment (czgu) [inline querybook/webapp/components/AppAdmin/components/DatePickerField/DatePickerField.tsx:23]: secondsToDate

File: querybook/webapp/resource/admin/announcement.ts [hunk 2]
```diff
@@ -1,18 +1,32 @@
 import { IAdminAnnouncement } from 'const/admin';
 import ds from 'lib/datasource';
+import moment from 'moment';
+
+function formatDate(date) {
```
Comment (czgu) [inline querybook/webapp/resource/admin/announcement.ts:5]: need typing for date

File: querybook/webapp/ui/Select/Select.tsx [hunk 3]
```diff
@@ -81,7 +82,7 @@ export function makeSelectOptions(options: IOptions) {
                     {option}
                 </option>
             ) : (
-                <option value={option.key} key={option.key}>
+                <option value={option.key} key={option.key} {...option}>
```
Comment (czgu) [inline querybook/webapp/ui/Select/Select.tsx:85]: hidden={option.hidden}

**Review by czgu — CHANGES_REQUESTED** [review]

File: querybook/webapp/components/AppAdmin/components/EnvironmentSelection/EnvironmentSelection.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+}) => {
+    const [field, , helper] = useField({ name });
+    const { values, setFieldValue } = useFormikContext<{
+        setFieldValue: () => void;
+        url_regex: string;
+    }>();
+
+    const value = options.find((o) => {
+        const regexPattern = new RegExp(values.url_regex);
+        return !!`/${o.value}/`.match(regexPattern);
```
[the remark this answers is not in the document]
Comment (alexVasylenko) [reply querybook/webapp/components/AppAdmin/components/EnvironmentSelection/EnvironmentSelection.tsx]: Oh, I tried to cover the next behavior. If a user selects any env, then he can edit the regex pattern and the selection should have the right value.

File: docs_website/docs/changelog/breaking_change.md [hunk 5]
```diff
@@ -10,3 +10,7 @@ Here are the list of breaking changes that you should be aware of when updating
 ## v2.8.0
 
 Result store plugin change. Now BaseReader::get_download_url requires a custom_name field to rename the download file. You only need to add the `custom_name=None` in the parameters for it to work.
+
+## v2.8.3
```
Comment (czgu) [inline docs_website/docs/changelog/breaking_change.md:14]: can you change the order (reverse chronological) and bump the version to 2.9.0?

File: plugins/webpage_plugin/custom_script.ts [hunk 6]
```diff
@@ -51,3 +58,44 @@ declare global {
         >;
     }
 }
+
+const NotificationItem = styled.div`
```
Comment (czgu) [inline plugins/webpage_plugin/custom_script.ts:62]: this is custom code?

File: querybook/webapp/ui/FormikField/DatePickerField.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    display: flex;
+    align-items: center;
+    width: 50%;
+`;
+
+const StyledDatePicker = styled(SearchDatePicker)`
+    margin-bottom: 0px;
+`;
+
+function secondsToDate(dateInSeconds?: number): number {
```
Comment (czgu) [inline querybook/webapp/ui/FormikField/DatePickerField.tsx:22]: Date is still seconds? so I suppose this function removes utcOffset?

File: package.json [hunk 8]
```diff
@@ -75,6 +75,7 @@
         "react": "17.0.2",
         "react-chartjs-2": "3.0.4",
         "react-codemirror2": "7.2.1",
+        "react-datepicker": "^4.3.0",
```
[the remark this answers is not in the document]
Comment (czgu) [reply package.json:78]: Can you remove the unused package?

**Review by czgu — COMMENTED** [review]
great just a couple minor comments before merge

File: querybook/webapp/ui/FormikField/DatePickerField.tsx (hunk 7 above)
Comment (alexVasylenko) [reply querybook/webapp/ui/FormikField/DatePickerField.tsx:22]: Yes, date is still in seconds. This function should restore date from UTC. I save the date in UTC but for user I want to show the time in UTC.

Comment (czgu) [reply querybook/webapp/ui/FormikField/DatePickerField.tsx:22]: rename it to utcSecondsToLocal?

File: querybook/migrations/versions/8b69979ffc31_add_expiry_date_for_announcements.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
+"""add expiry date for announcements
+
+Revision ID: 8b69979ffc31
+Revises: ea497b49195e
```
Comment (czgu) [inline querybook/migrations/versions/8b69979ffc31_add_expiry_date_for_announcements.py:4]: change to aceefd7708b6

File: querybook/migrations/versions/8b69979ffc31_add_expiry_date_for_announcements.py [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+Create Date: 2021-10-22 09:04:47.064570
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '8b69979ffc31'
+down_revision = 'ea497b49195e'
```
Comment (czgu) [inline querybook/migrations/versions/8b69979ffc31_add_expiry_date_for_announcements.py:14]: can you change down_revision to aceefd7708b6

**Review by czgu — COMMENTED** [review]
Added a couple minor things before we can merge since there is another db migration pr

**Review by czgu — APPROVED** [review]

**Pull request merged**